lib/jekyll/regenerator.rb

```diff
@@ -130,7 +130,7 @@ def add_dependency(path, dependency)
     #
     # Returns nothing.
     def write_metadata
-      File.open(metadata_file, 'w') do |f|
+      File.open(metadata_file, 'wb') do |f|
```

Comment:
Marshal is binary data, so this can crash on some contents :-( Sorry I missed that.